utils/nctl/sh/staging/stage.sh

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+
+source "$NCTL/sh/utils/main.sh"
+
+# ----------------------------------------------------------------
+# MAIN
+# ----------------------------------------------------------------
+
+#######################################
+# Initialises scenario binaries.
+# Arguments:
+#   Scenario ordinal identifier.
+#######################################
+function _set_assets()
+{
+    local PATH_TO_STAGING=${1}
+    local PATH_TO_STAGING_VERSION
+    local SCENARIO_COMMIT_HASH
+    local SCENARIO_VERSION
+    local STAGING_TARGET
+    local UPGRADE_STAGING_TARGET
+    local IFS
+
+    log "... setting assets"
+
+    pushd "$PATH_TO_STAGING/src" || exit
+
+    for UPGRADE_STAGING_TARGET in "${NCTL_STAGING_TARGETS[@]}"
+    do        
+        # Destructure settings.
+        IFS=':'
+        read -ra STAGING_TARGET <<< "$UPGRADE_STAGING_TARGET"
+        SCENARIO_COMMIT_HASH="${STAGING_TARGET[1]}"
+        SCENARIO_VERSION="${STAGING_TARGET[0]}"
+        PATH_TO_STAGING_VERSION="$PATH_TO_STAGING/$SCENARIO_VERSION"
+        log "... ... $SCENARIO_VERSION"
+
+        # Set commit.
+        git checkout "$SCENARIO_COMMIT_HASH" > /dev/null 2>&1
+
+        # Compile binaries.
+        # if [ "$NCTL_COMPILE_TARGET" = "debug" ]; then
+        #     cargo build --package casper-node
```

Comment:
It would probably be worth making use of the `CARGO_BUILD_TARGET_DIR` env var in order to have cargo build the artefacts to a directory named as per the short commit hash.  In that way, rebuilding any previously-built targets should be fast.

Setting the env var will work for raw `cargo build` commands and the `make build-contract-rs` calls, although the Makefile will need to be updated to have the `wasm-strip` command applied correctly in the `build-contract-rs` (can be done in a separate PR though).

For example:
```
CARGO_BUILD_TARGET_DIR="$NCTL_CASPER_HOME/target/$(git rev-parse --short HEAD)" cargo build --package casper-node
```

Reply:
I am minded to do this in a subsequent PR.